app/controllers/ops_controller/settings/common.rb

```diff
@@ -859,20 +859,26 @@ def settings_get_info(nodetype = x_node)
     nodes = nodetype.downcase.split("-")
     case nodes[0]
     when "root"
+      @subtab = -1
```

Comment:
hardcoding the numbers feels very brittle.  I'm wondering if there's a way that you can have a helper method in the SettingsTabsHelper.  I thinking something like

```ruby
SETTINGS_TAB_IDS = %w[
  settings_details
  settings_cu_collection
  settings_tags
  ...
]

def settings_tab_index_by_name(name)
  SETTINGS_TAB_IDS.index(name) || -1
end
```

and then caller in the controller would do something like

```ruby
@settings_tab = settings_tab_index_by_name(@sb[:active_tab])
```

Reply:
It's not perfect, but hopefully you get the idea in that the indexing is done with a proper array and a lookup - this way if a new tab is inserted or a tab is removed it just all works out without having to change anything.